File: MusicManagerAPI/src/main/java/com/ldz/music/manager/model/converter/beans/user/music/status/bo/to/dto/MusicTypeBOtoDTO.java
package com.ldz.music.manager.model.converter.beans.user.music.status.bo.to.dto;

import com.ldz.converter.container.annot.BeanConverter;
import com.ldz.converter.container.inter.IConverter;
import com.ldz.music.manager.model.MusicTypeDTO;
import com.ldz.music.manager.model.bo.MusicTypeBO;
import org.springframework.stereotype.Component;

/**
 * Created by ldalzotto on 26/04/2017.
 */
@Component
@BeanConverter(initialBeanClass = MusicTypeBO.class, targetBeanClass = MusicTypeDTO.class)
public class MusicTypeBOtoDTO implements IConverter<MusicTypeBO, MusicTypeDTO> {

    @Override
    public MusicTypeDTO apply(MusicTypeBO musicTypeBO) {
        MusicTypeDTO musicTypeDTO = new MusicTypeDTO();

        if (musicTypeBO != null) {
            musicTypeDTO.setSourceUrl(musicTypeBO.getSourceUrl());
            if (musicTypeBO.getMusicTypes() != null) {
                musicTypeDTO.setMusicType(musicTypeBO.getMusicTypes().name());
            }
        }

        return musicTypeDTO;
    }
}

File: MusicManagerAPI/src/scala/com/ldz/music/manager/model/converter/bo/to/dto/MusicTypeBOtoDTO.scala
package com.ldz.music.manager.model.converter.bo.to.dto

import com.ldz.converter.container.annot.BeanConverter
import com.ldz.converter.container.inter.IConverter
import com.ldz.music.manager.model.MusicTypeDTO
import com.ldz.music.manager.model.bo.MusicTypeBO
import org.springframework.stereotype.Component

/**
  * Created by Loic on 20/06/2017.
  */
@Component
@BeanConverter(initialBeanClass = classOf[MusicTypeBO], targetBeanClass = classOf[MusicTypeDTO])
class MusicTypeBOtoDTO extends IConverter[MusicTypeBO, MusicTypeDTO]{

  override def apply(musicTypeBO: MusicTypeBO): MusicTypeDTO = {
    musicTypeBO match {
      case `musicTypeBO` if musicTypeBO != null => MusicTypeDTO(musicTypeBO.getMusicTypes, musicTypeBO.getSourceUrl, musicTypeBO.getMusicParameters)
      case _ => null
    }
  }


}